shared/lib/wp/support.js

```diff
@@ -0,0 +1,102 @@
+export default function wpcomSupport( wpcom ) {
+	let supportUser = '';
+	let supportToken = '';
+
+	/**
+	 * Request parameters are as follows:
+	 * - param (required) A string or object that contains the path
+	 * - query (optional) An object that expands the response object
+	 * - body (required*) Required by POST and PUT, but not by GET and DEL
+	 * - fn (required) A callback function to handle the returned result
+	 * - ...rest (optional) Some queries have additional parameters after
+	 *   callback function
+	 *
+	 * Return the index of the query parameter, or if one does not exist,
+	 * return false.
+	 */
+	const getQueryIndex = function( req, args ) {
+		let fnIndex, queryIndex;
+
+		// Find the index of the callback in the arguments
+		fnIndex = args.findIndex( function( e ) {
+			return 'function' === typeof e;
+		} );
+
+		// Set queryIndex based on the request type and fnIndex
+		if ( req === 'post' || req === 'put' ) {
+			queryIndex = ( 3 === fnIndex ) ? 1 : false;
+		} else {
+			queryIndex = ( 2 === fnIndex ) ? 1 : false;
+		}
+		return queryIndex;
+	}
```

Comment:
There's a missing semicolon here, and with the other function assignments below.